client/pages/layout/layout.nav.controller.js

```diff
@@ -1,20 +1,18 @@
-angular.module('probedock.layout').controller('LayoutNavCtrl', function(api, appSettings, orgs, profile, $rootScope, $scope, $state) {
+angular.module('probedock.layout').controller('LayoutNavCtrl', function(api, appSettings, orgs, profile, $rootScope, $scope, states) {
 
+  orgs.forwardData($scope);
+  profile.forwardData($scope);
   appSettings.forwardSettings($scope);
 
-  var state = $state.current;
-  $rootScope.$on('$stateChangeSuccess', function(event, toState, toStateParams) {
-    state = toState;
-    $scope.orgName = toStateParams.orgName;
+  var currentStateName = null;
+  states.onStateChangeSuccess($scope, true, function(state, params, resolves) {
+    currentStateName = state.name;
   });
 
   $scope.baseStateIs = function() {
```

Comment:
Any function called from the views will be called for each digest cycle. Maybe it would be faster to listen to state transitions, store a `$scope.baseState` variable, and replace the call in the view by `baseState == "help"`. But I don't even know if it would actually be any faster.

Add a story if you want to check it out.